### PR TITLE
feat: [PIE-2792]: connector implementation and artifact and infrastructure ui preparation for azure integration

### PR DESCRIPTION
**Pull request by Tamaraius**
##### Summary:
Azure integration changes for Azure Connector, ACR Artifact and Azure Infra Definition

##### Jira Links:
https://harness.atlassian.net/browse/PIE-2873
https://harness.atlassian.net/browse/PIE-2794
https://harness.atlassian.net/browse/PIE-2792
https://harness.atlassian.net/browse/PIE-2791

Additional UI fixes noticed during UI demo are listed and will be done in https://harness.atlassian.net/browse/PIE-3258

##### Screenshots:
Connector
https://user-images.githubusercontent.com/96610348/160122800-e9b30501-dad6-4162-8ff5-298521e3e97d.mov

Sidecar Artifact
https://user-images.githubusercontent.com/96610348/160122700-5a6aed4a-bbec-4ec2-984b-9cd2cf81583e.mov

Primary Artifact
https://user-images.githubusercontent.com/96610348/160122629-567d047e-29e5-4c14-b16f-56c07b396af3.mov

Infrastructure
https://user-images.githubusercontent.com/96610348/160122732-30027ac6-1200-4047-adff-4ac1b14edcef.mov

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by Tamaraius** [comment]
retrigger titlecheck

**Review by PratyushGar — APPROVED** [review]

File: src/modules/35-connectors/components/CreateConnector/AzureConnector/StepAuth/AzureAuthentication.tsx [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+interface StepConfigureProps {
+  closeModal?: () => void
+  onSuccess?: () => void
+}
+
+interface AzureFormInterface {
+  authType: string | undefined
+  azureEnvironmentType: string | undefined
+  clientId: string | undefined
+  tenantId: string | undefined
```
Comment (bhavya-sinha) [inline src/modules/35-connectors/components/CreateConnector/AzureConnector/StepAuth/AzureAuthentication.tsx:58]: Can we move the interfaces to a separate interface file?

File: src/modules/35-connectors/pages/connectors/ConnectorsPage.tsx [hunk 2]
```diff
@@ -355,7 +356,16 @@ const ConnectorsPage: React.FC<ConnectorsListProps> = ({ catalogueMockData, stat
               },
               items:
                 item.connectors
-                  ?.filter(connector => (connector === 'ErrorTracking' ? isErrorTrackingEnabled : true))
+                  ?.filter(connector => {
+                    switch (connector) {
+                      case 'ErrorTracking':
+                        return isErrorTrackingEnabled
+                      case 'Azure':
+                        return isAzureEnabled
```
Comment (bhavya-sinha) [inline src/modules/35-connectors/pages/connectors/ConnectorsPage.tsx:364]: Can we move this piece of code to separate function?

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactUtils.ts [hunk 3]
```diff
@@ -253,6 +253,16 @@ export const defaultArtifactInitialValues = (selectedArtifact: ArtifactType): Im
         repositoryPortorRepositoryURL: RepositoryPortOrServer.RepositoryUrl
       }
     case ENABLED_ARTIFACT_TYPES.DockerRegistry:
+    case ENABLED_ARTIFACT_TYPES.Acr:
+      return {
+        identifier: '',
+        subscription: '',
+        registry: '',
+        repository: '',
```
Comment (bhavya-sinha) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactUtils.ts:261]: These need not be empty string for default values. We can just keep the fields which have some value by default, else yaml will be saved as empty string

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/K8sServiceSpec.tsx [hunk 4]
```diff
@@ -142,7 +142,12 @@ export class KubernetesServiceSpec extends Step<ServiceSpec> {
             includeAllConnectorsAvailableAtScope: true
           },
           body: {
-            types: [ArtifactToConnectorMap.DockerRegistry, ArtifactToConnectorMap.Gcr, ArtifactToConnectorMap.Ecr],
+            types: [
+              ArtifactToConnectorMap.DockerRegistry,
+              ArtifactToConnectorMap.Gcr,
+              ArtifactToConnectorMap.Ecr,
+              ArtifactToConnectorMap.Acr
```
Comment (bhavya-sinha) [inline src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/K8sServiceSpec.tsx:149]: What is this used for? I think we are not using this api call. CC: @chetanvekariya-harness

File: src/modules/35-connectors/pages/connectors/ConnectorsPage.tsx (hunk 2 above)
Comment (Tamaraius) [reply src/modules/35-connectors/pages/connectors/ConnectorsPage.tsx:364]: moved :)

File: src/modules/35-connectors/components/CreateConnector/AzureConnector/StepAuth/AzureAuthentication.tsx (hunk 1 above)
Comment (Tamaraius) [reply src/modules/35-connectors/components/CreateConnector/AzureConnector/StepAuth/AzureAuthentication.tsx:58]: moved :)

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactUtils.ts (hunk 3 above)
Comment (Tamaraius) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactUtils.ts]: removed :)

**Review by bhavya-sinha — APPROVED** [review]

**Review by PratyushGar — DISMISSED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Comment by samarth-sharma31** [comment]
retrigger jest

**Comment by Tamaraius** [comment]
Closing PR - Separate PRs will be created for every part - Connector, Artifact, Infrastructure

**Pull request closed without merging**